packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -172,47 +172,69 @@ function getEventMap(elm: EventTarget): ListenerMap {
     return listenerInfo;
 }
 
-const shadowRootEventListenerMap: WeakMap<EventListener, WrappedListener> = new WeakMap();
+// Using a WeakMap instead of a WeakSet because this one works in IE11 :(
+const eventsComingFromShadowRoot: WeakMap<Event, 1> = new WeakMap();
+
+function isEventComingFromShadowRoot(event: Event): boolean {
+    return eventsComingFromShadowRoot.has(event);
+}
+
+export function setEventFromShadowRoot(event: Event) {
+    eventsComingFromShadowRoot.set(event, 1);
+}
+
+const shadowRootEventListenerMap: WeakMap<
+    EventListenerOrEventListenerObject,
+    WrappedListener
+> = new WeakMap();
 
 function getWrappedShadowRootListener(
     sr: SyntheticShadowRootInterface,
-    listener: EventListener
+    listener: EventListenerOrEventListenerObject
```

Comment:
Shouldn't we also migrate `EventListener` to `EventListenerOrEventListenerObject` for the other listener wapper?